targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/spi_api.c

```diff
@@ -0,0 +1,363 @@
+                               (0ul <<  4) |    /* Motorola frame format            */
+                               (0ul <<  6) |    /* CPOL = 0                         */
+                               (0ul <<  7) |    /* CPHA = 0                         */
+                               (1ul <<  8) );   /* Set serial clock rate            */
+            obj->spi->CPSR  = (2ul <<  0);      /* set SSP clk to 6MHz (6.6MHz max) */
+            obj->spi->CR1   = ((1ul <<  1) |    /* Synchronous serial port enable   */
+                               (0ul <<  2) );   /* Device configured as master      */
+            break;
+        case (int)SPI_2:
+            obj->spi->CR1   = 0;
+            obj->spi->CR0   = SSP_CR0_SCR_DFLT | SSP_CR0_FRF_MOT | SSP_CR0_DSS_8;
+            obj->spi->CPSR  = SSP_CPSR_DFLT;
+            obj->spi->IMSC  = 0x8;
+            obj->spi->DMACR = 0;
+            obj->spi->CR1   = SSP_CR1_SSE_Msk;
+            obj->spi->ICR   = 0x3;
+            break;
+        case (int)SPI_3:
+            obj->spi->CR1   = 0;
+            obj->spi->CR0   = SSP_CR0_SCR_DFLT | SSP_CR0_FRF_MOT | SSP_CR0_DSS_8;
+            obj->spi->CPSR  = SSP_CPSR_DFLT;
+            obj->spi->IMSC  = 0x8;
+            obj->spi->DMACR = 0;
+            obj->spi->CR1   = SSP_CR1_SSE_Msk;
+            obj->spi->ICR   = 0x3;
+            break;
+        case (int)SPI_4:
+            obj->spi->CR1   = 0;
+            obj->spi->CR0   = SSP_CR0_SCR_DFLT | SSP_CR0_FRF_MOT | SSP_CR0_DSS_8;
+            obj->spi->CPSR  = SSP_CPSR_DFLT;
+            obj->spi->IMSC  = 0x8;
+            obj->spi->DMACR = 0;
+            obj->spi->CR1   = SSP_CR1_SSE_Msk;
+            obj->spi->ICR   = 0x3;
+            break;
+    }
+
+    if (mosi != NC) {
+        altfunction[0] = 1;
+    } else {
+        altfunction[0] = 0;
+    }
+    if (miso != NC) {
+        altfunction[1] = 1;
+    } else {
+        altfunction[1] = 0;
+    }
+    if (sclk != NC) {
+        altfunction[2] = 1;
+    } else {
+        altfunction[2] = 0;
+    }
+    if (ssel != NC) {
+        altfunction[3] = 1;
+    } else {
+        altfunction[3] = 0;
+    }
+
+    /* Enable alt function */
+    switch ((int)obj->spi) {
+        case (int)SPI_2:    /* Shield ADC SPI */
+            GPIO_MAP[CMSDK_GPIO_ADC_MOSI_SPI_GPIO_NUM]->ALTFUNCSET |=
+                        (altfunction[0]<<CMSDK_GPIO_ALTFUNC_ADC_MOSI_SPI_SET);
+            GPIO_MAP[CMSDK_GPIO_ADC_MISO_SPI_GPIO_NUM]->ALTFUNCSET |=
+                        (altfunction[1]<<CMSDK_GPIO_ALTFUNC_ADC_MISO_SPI_SET);
+            GPIO_MAP[CMSDK_GPIO_ADC_SCK_SPI_GPIO_NUM]->ALTFUNCSET |=
+                        (altfunction[2]<<CMSDK_GPIO_ALTFUNC_ADC_SCK_SPI_SET);
+            GPIO_MAP[CMSDK_GPIO_ADC_CS_SPI_GPIO_NUM]->ALTFUNCSET |=
+                        (altfunction[3]<<CMSDK_GPIO_ALTFUNC_ADC_CS_SPI_SET);
+            break;
+        case (int)SPI_3:   /* Shield 0 SPI */
+            GPIO_MAP[CMSDK_GPIO_SH0_MOSI_SPI_GPIO_NUM]->ALTFUNCSET |=
```

Comment:
@0xc0170  This functionalty belongs to pin_fucntion and not to pin_mode. pin_mode is not supported by the target.